gsy_framework/sim_results/electric_blue/time_series.py

```diff
@@ -0,0 +1,148 @@
+from typing import Callable, Dict
+
+from pendulum import DateTime, Duration
+
+from gsy_framework.enums import AvailableMarketTypes
+from gsy_framework.forward_markets.forward_profile import (
+    ForwardTradeProfileGenerator)
+from gsy_framework.sim_results.electric_blue.aggregate_results import (
+    ForwardDeviceStats)
+
+
+def resample_data(
+        timeseries_data: Dict[DateTime, float],
+        resolution: Duration, aggregator_fn: Callable) -> Dict:
+    """Aggregate data using the specified resolution and aggregator function."""
+
+    result = {}
+
+    sorted_time_slots = sorted(timeseries_data.keys())
+    start_time = sorted_time_slots[0]
+    next_time = start_time + resolution
+
+    time_slots_to_aggregate = []
+    for time_slot in sorted_time_slots:
+        if time_slot >= next_time:
+            result[time_slots_to_aggregate[0]] = aggregator_fn(
```

Comment:
I keep on wondering about the possibility to need a different aggregation function, however does not hurt to leave it that way in order to have that possibility in the future.